src/world/Units/Creatures/AIInterface.h

```diff
@@ -107,538 +88,334 @@ enum AI_SpellTargetType
+    bool isTargetingDisabled() { return mIsTargetingDisabled; }
 
     //////////////////////////////////////////////////////////////////////////////////////////
-    // AI Script functions
-    private:
-
-        AiScriptTypes mAiScriptType;
-
-    public:
-
-        void setAiScriptType(AiScriptTypes ai_type) { mAiScriptType = ai_type; }
-        uint32_t getAiScriptType() { return mAiScriptType; }
-        bool isAiScriptType(AiScriptTypes ai_type) { return ai_type == mAiScriptType; }
+    // Misc functions
+    void Init(Unit* un, AiScriptTypes at);
+    void Init(Unit* un, AiScriptTypes at, Unit* owner);   // used for pets
+    Unit* getUnit() const;
+    Unit* getPetOwner() const;
+    Unit* getCurrentTarget() const;
+    LocationVector m_lasttargetPosition;
+
+    bool isGuard() { return m_isNeutralGuard; }
+    void setGuard(bool value) { m_isNeutralGuard = value; }
+    void setCurrentTarget(Unit* pUnit) { m_target = pUnit; }
+    float calcCombatRange(Unit* target, bool ranged);
+
+    void updateEmotes(unsigned long time_passed);
+    void eventAiInterfaceParamsetFinish();
+    TimedEmoteList* timed_emotes;
+
+    bool moveTo(float x, float y, float z, float o = 0.0f, bool running = false);
+    void calcDestinationAndMove(Unit* target, float dist);
+
+    // boundary system methods
+    bool checkBoundary();
+    CreatureBoundary const getBoundary() const { return _boundary; }
+    void addBoundary(AreaBoundary const* boundary, bool overrideDefault = false, bool negativeBoundaries = false);
+    void setDefaultBoundary();
+    void clearBoundary();
+    static bool isInBounds(CreatureBoundary const* boundary, LocationVector who);
+    bool isInBoundary(LocationVector who) const;
+    bool isInBoundary() const;
+    void doImmediateBoundaryCheck() { m_boundaryCheckTime.resetInterval(0); }
+
+    bool canUnitEvade(unsigned long time_passed);
+    void enterEvadeMode();
+    bool _enterEvadeMode();
+
+    void initGroupThreat(Unit* target);
+    void instanceCombatProgress(bool activate);
+
+    // Event Handler
+    void HandleEvent(uint32_t event, Unit* pUnit, uint32_t misc1);
+
+    void EventForceRedirected(Unit* pUnit, uint32_t misc1);
+    void EventHostileAction(Unit* pUnit, uint32_t misc1);
+    void EventUnitDied(Unit* pUnit, uint32_t misc1);
+    void EventUnwander(Unit* pUnit, uint32_t misc1);
+    void EventWander(Unit* pUnit, uint32_t misc1);
+    void EventUnfear(Unit* pUnit, uint32_t misc1);
+    void EventFear(Unit* pUnit, uint32_t misc1);
+    void EventFollowOwner(Unit* pUnit, uint32_t misc1);
+    void EventDamageTaken(Unit* pUnit, uint32_t misc1);
+    void EventLeaveCombat(Unit* pUnit, uint32_t misc1);
+    void EventEnterCombat(Unit* pUnit, uint32_t misc1);
+    void EventChangeFaction(Unit* ForceAttackersToHateThisInstead = NULL);    /// we have to tell our current enemies to stop attacking us, we should also forget about our targets
+    void OnDeath(Object* pKiller);
```

Comment:
Functions should start with lowercase character (includes all event functions above)